pageserver/src/tenant/timeline/walreceiver/walreceiver_connection.rs

```diff
@@ -140,7 +144,11 @@ pub(super) async fn handle_walreceiver_connection(
                 _ = connection_cancellation.cancelled() => info!("Connection cancelled"),
             }
             Ok(())
-        },
+        }
+        // Enrich the log lines emitted by this closure with meaningful context.
+        // TODO: technically, this task outlives the surrounding function, so, the
+        // spans won't be properly nested.
+        .in_current_span(),
```

Comment:
this looks a little bit suspicious because if we are viewing the spans from opentelemetry collectors (i.e., jaeger), there will be a super long top level span (whereas spans is usually per-request and short-lived); also some visualizer will not display child spans until the parent span is finished. But as long as we are using this for logging purpose, it should be fine.

Reply:
The connection_cancellation is tied to the surrounding ding function, so, this task generally doesn’t live much longer thank the surrounding function invocation.

so, it’s fine in this case, it’s not actually a long-running task

generally, I’d agree though